components/plugin_marketplace/marketplace_item/marketplace_item.tsx

```diff
@@ -70,114 +75,134 @@ export type MarketplaceItemProps = {
 
     error?: string;
 
-    button: JSX.Element;
+    controls: JSX.Element;
     updateDetails: JSX.Element | null;
     versionLabel: JSX.Element| null;
 };
 
-export default class MarketplaceItem extends React.PureComponent <MarketplaceItemProps> {
-    render(): JSX.Element {
-        let icon;
-        if (this.props.iconSource) {
-            icon = (
-                <div className='icon__plugin icon__plugin--background'>
-                    <img src={this.props.iconSource}/>
-                </div>
-            );
-        } else {
-            icon = <PluginIcon className='icon__plugin icon__plugin--background'/>;
-        }
-
-        let labels;
-        if (this.props.labels && this.props.labels.length !== 0) {
-            labels = this.props.labels.map((label) => (
-                <Label
-                    key={label.name}
-                    name={label.name}
-                    description={label.description}
-                    url={label.url}
-                    color={label.color}
-                />
-            ),
-            );
-        }
-
-        const pluginDetailsInner = (
-            <>
-                {this.props.name}
-                {this.props.versionLabel}
-            </>
+export default function MarketplaceItem(props: MarketplaceItemProps) {
+    const intl = useIntl();
+
+    const isCloudLicense = useSelector(isCurrentLicenseCloud);
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+    const limits = useSelector(getCloudLimits);
+
+    let icon;
+    if (props.iconSource) {
+        icon = (
+            <div className='icon__plugin icon__plugin--background'>
+                <img src={props.iconSource}/>
+            </div>
         );
+    } else {
+        icon = <PluginIcon className='icon__plugin icon__plugin--background'/>;
+    }
 
-        const description = (
-            <p className={classNames('more-modal__description', {error_text: this.props.error})}>
-                {this.props.error ? this.props.error : this.props.description}
-            </p>
+    let labels: React.ReactNodeArray = [];
+    if (props.labels && props.labels.length !== 0) {
+        labels = props.labels.map((label) => (
+            <Label
+                key={label.name}
+                name={label.name}
+                description={label.description}
+                url={label.url}
+                color={label.color}
+            />
+        ));
+    }
+
+    if (isCloudFreeEnabled && isCloudLicense && limits?.integrations?.enabled && Constants.Integrations.FREEMIUM_USAGE_IGNORED_PLUGINS.includes(props.id)) {
```

Comment:
nit: for this kind of conditions is hard to read exactly what the intention of the validation is. It is a good practice to document the code with variable names that makes sense and express intention, like `const thisAVariableNameThatExpressIntention = isCloudFreeEnabled && isCloudLicense && limits?.integrations?.enabled && Constants.Integrations.FREEMIUM_USAGE_IGNORED_PLUGINS.includes(props.id)` and then use the variable in the if condition. That makes code easier to read and it remains better in time, even better than comment lines, normally people updates code but not comments.